js_modules/dagit/packages/core/src/ticks/EvaluateSchedule.tsx

```diff
@@ -0,0 +1,400 @@
+            >
+              {selectedTimestampRef.current.label}
+            </Button>
+          </div>
+        </Popover>
+      </SelectWrapper>
+    );
+  }, [
+    _selectedTimestamp,
+    containerProps,
+    data,
+    isTickSelectionOpen,
+    jobName,
+    name,
+    repoAddress,
+    shouldEvaluate,
+    userTimezone,
+    viewport.width,
+  ]);
+
+  const buttons = React.useMemo(() => {
+    if (!shouldEvaluate) {
+      return (
+        <>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button
+            data-testid={testId('evaluate')}
+            intent="primary"
+            onClick={() => {
+              setShouldEvaluate(true);
+            }}
+          >
+            Evaluate
+          </Button>
+        </>
+      );
+    } else {
+      return <Button onClick={onClose}>Close</Button>;
+    }
+  }, [onClose, shouldEvaluate]);
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      style={{width: '70vw', display: 'flex'}}
+      title={
+        <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+          <Icon name="schedule" />
+          <span>{name}</span>
+        </Box>
+      }
+    >
+      <DialogBody>
+        <div style={{minHeight: '300px'}}>{content}</div>
+      </DialogBody>
+      {buttons ? <DialogFooter topBorder>{buttons}</DialogFooter> : null}
+    </Dialog>
+  );
+};
+
+export const GET_SCHEDULE_QUERY = gql`
+  query getSchedule(
+    $scheduleSelector: ScheduleSelector!
+    $startTimestamp: Float
+    $ticksAfter: Int
+    $ticksBefore: Int
+  ) {
+    scheduleOrError(scheduleSelector: $scheduleSelector) {
+      __typename
+      ... on PythonError {
+        message
+        stack
+      }
+      ... on Schedule {
+        name
+        potentialTickTimestamps(
+          startTimestamp: $startTimestamp
+          upperLimit: $ticksAfter
+          lowerLimit: $ticksBefore
+        )
+      }
+    }
+  }
+`;
+
+const EvaluateScheduleContent: React.FC<{
+  repoAddress: RepoAddress;
+  name: string;
+  timestamp: number;
+  jobName: string;
+}> = ({repoAddress, name, timestamp, jobName}) => {
+  const [userTimezone] = React.useContext(TimezoneContext);
+  const [scheduleDryRunMutation] = useMutation<
+    ScheduleDryRunMutation,
+    ScheduleDryRunMutationVariables
+  >(
+    SCHEDULE_DRY_RUN_MUTATION,
+    React.useMemo(() => {
+      const repositorySelector = repoAddressToSelector(repoAddress);
+      return {
+        variables: {
+          selectorData: {
+            ...repositorySelector,
+            scheduleName: name,
+          },
+          timestamp,
+        },
+      };
+    }, [name, repoAddress, timestamp]),
+  );
+  const [result, setResult] = React.useState<Awaited<
+    ReturnType<typeof scheduleDryRunMutation>
+  > | null>(null);
+  React.useEffect(() => {
+    scheduleDryRunMutation().then((result) => {
+      setResult(() => result);
+    });
+  }, [scheduleDryRunMutation]);
```

Comment:
It's a little weird that something called `EvaluateScheduleContent` immediately invokes a mutation as soon as it's mounted. Maybe this should have a name that indicates it's the end of the workflow like `DryRunResultsContent` rather than just the name of the modal.

Reply:
Yeah, I mostly just didn't want the parent component to call the mutation and save the data and then pass it down to this component because the parent component already has lots of state. I was mostly just testing around to see, I can switch it back but really the difference between a query/mutation is... kind of arbitrary tbh. In this case we're not persisting any information, it's effectively a query, where as other queries do logging and other kinds of stuff which do persist data.